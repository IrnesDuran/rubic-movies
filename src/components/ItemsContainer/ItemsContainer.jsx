import React from 'react';
import ItemCard  from '../ItemCard/ItemCard';
import { withRouter } from 'react-router-dom';
import './ItemsContainer.styles.scss';

const ItemsContainer =(props) => {
    return(
        <div className='items-container'>
            {
                props.data.filter((item, idx) => idx < 10)
                .map((item) => (<div className="item" key={item.id} onClick={() => props.history.push(`${props.searchCategory}/${item.id}`)}><ItemCard item={item}/></div>))
            }
        </div>
    )};

export default withRouter(ItemsContainer);