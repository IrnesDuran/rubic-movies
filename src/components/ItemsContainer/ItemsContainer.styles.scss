.items-container{
    display: flex;
    flex-wrap: wrap;
    align-content: flex-start;
    justify-content: center;
}

.item {
    width: 45%;
}



@media screen and (max-width: 800px) {
    .item {
      width:100%;
    }
}

