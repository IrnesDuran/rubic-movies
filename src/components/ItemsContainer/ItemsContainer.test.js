import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import ItemsContainer from './ItemsContainer';


describe ("ItemsContainer component test",()=> {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ItemsContainer/>);

    });


    it('should test ItemsContainer component snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    });

})