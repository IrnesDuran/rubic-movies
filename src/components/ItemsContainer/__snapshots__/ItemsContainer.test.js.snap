// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsContainer component test should test ItemsContainer component snapshot 1`] = `ShallowWrapper {}`;
